### PR TITLE
Ipam

### PR DESCRIPTION
**Pull request by tomdee**

### CODE REVIEW

**Comment by tomdee** [comment]
@robbrockbank Could you take a look at this? Still work to do on the tests but would be good for you to take a look.

**Comment by robbrockbank** [comment]
LGTM, I don't have anything.

Obviously there are a few tweaks required when we have actual releases of stuff, but otherwise this looks good.

**Comment by tomdee** [comment]
@robbrockbank I plan to merge this as soon as I get the tests passing. Do you want to take a final look before I do?

File: Makefile [hunk 1]
```diff
@@ -87,10 +93,10 @@ run-consul:
 
 create-dind:
 	@echo "You may want to load calico-node with"
-	@echo "docker load --input /code/calico-node.tar"
+	@echo "docker load --input /code/calico-node.tgz"
 	@ID=$$(docker run --privileged -v `pwd`:/code \
 	-e DOCKER_DAEMON_ARGS=--cluster-store=consul://$(LOCAL_IP_ENV):8500 \
-	-tid tomdee/dind-ux) ;\
+	-tid tomdee/dind-ipam) ;\
```
Comment (robbrockbank) [inline Makefile]: do we have a non tomdee release of this?

File: libnetwork/driver_plugin.py [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+    app.logger.debug("CreateNetwork JSON=%s", json_data)
+
+    # Create the CNM "network" as a Calico profile.
+    network_id = json_data["NetworkID"]
+    app.logger.info("Creating profile %s", network_id)
+    client.create_profile(network_id)
+
+    # Create a calico Pool for the CNM pool that was passed in.
+    for version in (4, 6):
+        if json_data["IPv%sData" % version]:
```
Comment (robbrockbank) [inline libnetwork/driver_plugin.py]: minor niggle:  would be nicer to just get the ip data once, e.g.
ip_data = json_data[blah]
if ip_data then do stuff with ip_data

File: tests/st/utils/workload.py [hunk 3]
```diff
@@ -50,30 +51,24 @@ def __init__(self, host, name, image="busybox", network=None):
...
             if network is not NET_NONE:
                 assert isinstance(network, DockerNetwork)
-            args.append("--net %s" % network)
-        args.append(image)
-        command = ' '.join(args)
+            network_command = "--net %s" % network
 
-        logger.info("Creating workload\n%s", command)
+        command = "docker run -tid --name %s %s %s" % (name,
+                                                          network_command,
```
Comment (robbrockbank) [inline tests/st/utils/workload.py]: minor alignment issue

**Comment by robbrockbank** [comment]
LGTM (very minor comments)

File: Makefile (hunk 1 above)
Comment (tomdee) [inline Makefile]: I'm sorting it out now. I might merge first then fix this up.

**Pull request merged**